src/overlays/actors/ovl_En_Tubo_Trap/z_en_tubo_trap.c

```diff
@@ -21,26 +27,324 @@ const ActorInit En_Tubo_Trap_InitVars = {
     (ActorFunc)EnTuboTrap_Update,
     (ActorFunc)EnTuboTrap_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/ovl_En_Tubo_Trap_0x809307E0/EnTuboTrap_Init.asm")
+s32 D_8093145C[] = { // actorprocessinitchain
+    0xC85800C5,
+    0xB0FC0FA0,
+    0xB1000064,
+    0x31040064,
+};
+
+// used in WaitForProximity
+s8 D_8093146C[] = {
+    0x28, 0x28, 0x28, 0x19,
+    0x1C, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00,
+};
+
+void EnTuboTrap_Init(Actor *thisx, GlobalContext *globalCtx) {
+    EnTuboTrap* this = THIS;
+
+    Actor_ProcessInitChain(thisx, &D_8093145C);
+    thisx->shape.rot.z = 0;
```

Comment:
thisx should be avoided for matching when possible, which is like 99.999% of the time (for example, this one should be `this->actor.shape`

Reply:
Yeah I've been trying to avoid accidentally using thisx when it could be actor, it just slips past me sometimes because this and thisx are too alike visually. I'll take another pass